### ISSUE TITLE
Why require Nitro?

### ISSUE THREAD

**Issue by marbemac** [nksaraf/vinxi#4]
Just got done catching up on Ryan's last stream and love what you're trying to do here with Vinxi!

> The JavaScript Server SDK to build full stack apps and frameworks with your own opinions.

💯, but curious if there's any reason Vinxi can't leave the server router/runtime decision to the end user as well? E.g. if the end user wants to use Hono, or Bun/Elysia, etc, rather than nitro/h3 eventHandler style approach to handling incoming server requests and streaming back the reac/solid/etc app.

---

Side note, was playing around with the examples and noticed the typings/intellisense (at least in vscode) isn't working in many cases. Adjusting the root tsconfig.json#moduleResolution to `bundler` enables support for package.json export fields and fixes the issues :).

<img width="1254" alt="Screenshot 2023-09-26 at 9 58 53 AM" src="https://github.com/nksaraf/vinxi/assets/847542/e031d649-4656-498e-aadb-cb33f9b38a95">

**Comment by nksaraf** [comment]
> if the end user wants to use Hono, or Bun/Elysia, etc, rather than nitro/h3 eventHandler style approach to handling incoming server requests and streaming back the reac/solid/etc app.

I haven't explored this approach yet. But you are right, in a sense nitro is built of two parts. The rollup based bundler for the server that can make our servers work in any environment. And a wrapper around `h3` for the server runtime. `h3` takes care of a lot of problems with runtime differences which is really valuable. you can build any pattern on top and different server APIs. So something like Hono's API can be easily emulated by h3 if thats desired. Supporting too many tools in a zone which doesn't really gain much by having opinions prevents deeper integration.

**Comment by apatrida** [comment]
It looks like the dev server is also hard coded here to one model in Nitro instead of allowing adaptable dev servers.  I.e. for CloudFlare using Wrangler isn't possible is it, due to the hard coded Vinx dev server code only having one model.

(see also: https://nitro.unjs.io/deploy/providers/cloudflare)

Or if I wanted to run this as SSR inside GraalVM from a JVM server and have an adapter to use this as middleware and I handle all file system activities, the dev server breaks this all again due to a hard coded single model.

The dev server is important for adapters as much as the production mode is.

**Comment by nksaraf** [comment]
> The dev server is important for adapters as much as the production mode is.

I completely agree. I think the dev server almost needs to be environment adaptive. Right now I have left enough room for you do anything in the event handler including using a miniflare instance to run the handler code. 

I haven't figure out all the entry points needed for an adaptive dev server and what use cases I want to serve. But you are right that right now its a node dev server only and might work in bun but the dev server is not adaptive. I am keen on hearing ideas about what kind of stuff you would need it to do as dev server in those mentioned environments

**Comment by apatrida** [comment]
Our goal is using this within a GraalVM running a server like Spring Boot or Quarkus serving and handling the traffic for the whole system, dealing with all api endpoints, and then delegating some paths to the vinxi based server (solid start in our case) both in dev and production modes.

**Comment by peterhirn** [comment]
So if I want to upgrade my Cloudflare deployed SolidStart Beta 1 app to Beta 2 I have to write my own Wrangler/dev-server integration?

A Node.js dev-server will not work for me since I'm using KV/D1/etc.

Any pointers on how I can get started on this? Right now I don't even know how to deploy Beta 2 apps to Cloudflare.

**Comment by nksaraf** [comment]
I have an example working with miniflare in the Vinxi repo.. it's not a formalized integration but it works. It uses the older miniflare not the one based on workerd coz I need it to be able to communicate with the node runtime.
